test/e2e/shutdown/shutdown_test.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+package shutdown
+
+import (
+	"flag"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/open-policy-agent/opa/test/e2e"
+)
+
+var testRuntime *e2e.TestRuntime
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	testServerParams := e2e.NewAPIServerTestParams()
+
+	testServerParams.ShutdownWaitPeriod = 10
```

Comment:
10 seconds is maybe a little bit long, can we lower this (even down to like 1 second?) and use a smaller sleep time in the test? In addition we should explicitly set the graceful shutdown period to 0 or something less than the wait time to ensure the delay is from the wait period.

Reply:
@patrick-east thanks for you feedback! I changed `ShutdownWaitPeriod` to 2 seconds and `GracefulShutdownPeriod` to 1 second while sleeping for 1500ms. If I set `GracefulShutdownPeriod` to 0, part of the test would fail with `exit 1`.